src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -57,9 +62,18 @@ public override async Task UpdateCSharpBufferAsync(JToken token, CancellationTok
                 throw new ArgumentNullException(nameof(token));
             }
 
-            await _joinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            await _updateCSharpSemaphoreSlim.WaitAsync();
```

Comment:
Are we sure that there isn't a race up until this point in the JsonRpc bits?

Reply:
It appears so with whatever testing I did with Debug.Fail. It was never hit after these changes both before and after switching to the main thread. Is there a better way to make sure of that?

Reply:
> Is there a better way to make sure of that?

Aside from looking at their code a good "ensurement" would be to add a for-loop on the server and send the update buffer request 100k times with an incrementing version and ensure on the client that they come in the right order.

Reply:
After some pair debugging with @NTaylorMullen, we realized this isn't a complete fix and the real fix should come from the VS Language Client implementation. Going to merge this as is because this is still a good mitigation and will start a thread to with the VS LSP Client team.